crates/settlement-clients/ethereum/src/conversion.rs

```diff
@@ -1,12 +1,304 @@
+use alloy::dyn_abi::parser::Error;
+use alloy::eips::eip4844::BYTES_PER_BLOB;
 use alloy::primitives::U256;
+use alloy_primitives::FixedBytes;
+use c_kzg::{Blob, KzgCommitment, KzgProof, KzgSettings};
+use color_eyre::{eyre::ContextCompat, Result as EyreResult};
+use std::fmt::Write;
 
 /// Converts a `&[Vec<u8>]` to `Vec<U256>`. Each inner slice is expected to be exactly 32 bytes long.
 /// Pads with zeros if any inner slice is shorter than 32 bytes.
-pub(crate) fn slice_slice_u8_to_vec_u256(slices: &[[u8; 32]]) -> Vec<U256> {
+pub(crate) fn vec_u8_32_to_vec_u256(slices: &[[u8; 32]]) -> EyreResult<Vec<U256>> {
     slices.iter().map(|slice| slice_u8_to_u256(slice)).collect()
 }
 
 /// Converts a `&[u8]` to `U256`.
-pub(crate) fn slice_u8_to_u256(slice: &[u8]) -> U256 {
-    U256::try_from_be_slice(slice).expect("could not convert u8 slice to U256")
+pub(crate) fn slice_u8_to_u256(slice: &[u8]) -> EyreResult<U256> {
+    U256::try_from_be_slice(slice).wrap_err_with(|| "could not convert &[u8] to U256".to_string())
+}
+
+// Function to convert a slice of u8 to a padded hex string
+// Function only takes a slice of length up to 32 elements
+// Pads the value on the right side with zeros only if the converted string has lesser than 64 characters.
```

Comment:
Use docstrings to document functions (`///` instead of `//`).

Reply:
Valid, updated.